lib/SQL/Translator/Parser/SQLite.pm

```diff
@@ -425,7 +425,7 @@ table_constraint : PRIMARY_KEY parens_field_list conflict_clause(?)
         }
     }
     |
-    FOREIGN_KEY parens_field_list REFERENCES ref_def cascade_def(?)
+    FOREIGN_KEY parens_field_list REFERENCES ref_def cascade_def(?) deferrable(?) deferred(?)
```

Comment:
I've updated the PR accordingly. All tests are passing.

On Wed, Sep 16, 2015 at 7:12 PM, Dagfinn Ilmari Mannsåker <
notifications@github.com> wrote:

> In lib/SQL/Translator/Parser/SQLite.pm
> <https://github.com/dbsrgits/sql-translator/pull/70#discussion_r39672522>:
>
> > @@ -425,7 +425,7 @@ table_constraint : PRIMARY_KEY parens_field_list conflict_clause(?)
> >          }
> >      }
> >      |
> > -    FOREIGN_KEY parens_field_list REFERENCES ref_def cascade_def(?)
> > +    FOREIGN_KEY parens_field_list REFERENCES ref_def cascade_def(?) deferrable(?) deferred(?)
>
> Unlike PostgreSQL, the INITIALLY (DEFERRED|IMMEDIATE) clause in SQLite is
> part of the (NOT)? DEFERRABLE clause. See
> https://sqlite.org/syntax/foreign-key-clause.html.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dbsrgits/sql-translator/pull/70/files#r39672522>.
>



-- 
Thanks,
Rob Kinyon
http://greenfishbluefish.com/